my_game.py

```diff
@@ -128,14 +129,36 @@ def thrust(self):
             self.change_x *= player_x_and_y_speed_ratio
             self.change_y *= player_x_and_y_speed_ratio
 
-    def update(self):
+    def reset(self):
+        self.reset_timer = PLAYER_INVINCIBILTY_SECONDS
+        # The Player is Invisible
+        self.alpha = 0
+
+    @property
+    def invicible(self):
+        return self.reset_timer < 1
+
+    def update(self, delta_time):
         """
         Move the sprite and wrap
         """
 
         self.center_x += self.change_x
         self.center_y += self.change_y
 
+        # Time when you can't get hit by an asteroid
+        if self.reset_timer > 0:
+            self.reset_timer -= delta_time
+            # Time when you are not visible
+            if self.reset_timer < 3:
```

Comment:
How about this? I think that's easier to understand.
```
if self.reset_timer < 3 and self.alpha == 0:
```

Also, add a comment about what this code accomplishes.